pulsar-client-cpp/lib/ConsumerConfigurationImpl.h

```diff
@@ -31,13 +31,18 @@ struct ConsumerConfigurationImpl {
     int receiverQueueSize;
     std::string consumerName;
     long brokerConsumerStatsCacheTimeInMs;
+    CryptoKeyReader* cryptoKeyReader;
```

Comment:
Can we use shared pointer instead like we do for MessageRouter

Reply:
Slightly different use case here. It has a reference to the application allocated implementation of CryptoKeyReader